optaweb-employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterGenerator.java

```diff
@@ -646,10 +496,24 @@ public RosterState createRosterState(GeneratorType generatorType, Tenant tenant,
                                                                         endDayOffset, endTime, rotationEmployee);
                         entityManager.persist(shiftTemplate);
                         shiftTemplateList.add(shiftTemplate);
+
+                        if (spot.getName().endsWith("Nurses")) {
+                            rotationEmployeeIndex = generatorType.rotationEmployeeIndexCalculator
+                                    .apply(startDayOffset, timeslotRangesIndex).get(1);
+                            rotationEmployee = rotationEmployeeIndex >= rotationEmployeeList.size() ? null :
+                                    rotationEmployeeList.get(rotationEmployeeIndex);
+                            shiftTemplate = new ShiftTemplate(tenantId, spot, startDayOffset, startTime,
+                                                              endDayOffset, endTime, rotationEmployee);
+                            entityManager.persist(shiftTemplate);
+                            shiftTemplateList.add(shiftTemplate);
+                        }
                     }
                 }
             }
-        }
+        };
+        // Create COVID templates first as they are more limited
+        spotList.stream().filter(s -> s.getName().startsWith("COVID")).forEach(createShiftTemplatesForWard);
```

Comment:
Since you're referencing the `COVID` prefix in a lot of places, I'd like to have it be a constant and reused. Avoids bugs.